docs/testing-styleguide.MD

```diff
@@ -0,0 +1,278 @@
+
+#### Asserting Styles
+All branches, paths, props and each of their conditions need to be tested to ensure 100% coverage. Where props are being used to conditionally apply style to a component, we use the assertStyle method to test that all of the CSS properties in the parameters have been applied. For example,
+
+```
+describe('FlatTableRow', () => {
+  it('then the element should have proper outline when focused', () => {
+      assertStyleMatch({
+        outline: `2px solid ${baseTheme.colors.focus}`,
+        outlineOffset: '-1px'
+      }, wrapper, { modifier: ':focus' });
+    });
+});
+```
+
+#### Snapshots
+Snapshots render a string output of what the DOM receives. The first time the test is run, a snapshot file is generated and stored in the `__snapshots__` directory within each component in the `/src/components` directory. The next time a test is run, if nothing has changed, the test passes. However, if we change something, the test fails and a diff is indicated in the output. You can update snapshots easily using the `jest -u` command although you must take due care and consideration to ensure the changes are correct before simply updating them.
+
+You should use snapshots in tests if amendments are being made to non-styled components to check styling is applied, but where existing and new styled components are concerned the use of snapshots is left up to the developer to decide if there is value. If you do want to use them, ensure they are small, focussed and effective.
+
+#### Continuous Integration (CI)
+The Travis CI service runs unit tests at every commit/push/pull request creation. You can manually run the tests with:
+1.	`npm run lint` - runs lint test.
+2.	`npm test` - runs jest tests.
+3.	Results can be investigated from the terminal or from the Travis dashboard linked from the branch/pull request checks
+
+### Functional Browser Testing
+#### Manual Testing
+Every new or amended component should be fully tested in either Storybook and/or [codesandbox.io](https://codesandbox.io) using functional testing, UI testing and internationalization testing techniques. We use [Jira](https://www.atlassian.com/software/jira) for issue and project tracking. 
+
+##### Browser Support
+Carbon is supported in the following browsers:
+* Chrome
+* Firefox
+* Safari
+* Edge Chromium
+
+Components should be tested in all supported browsers unless the piece of functionality is not specifically related to a UI feature, e.g. decimal precision.
+
+##### Jira
+For every testable task, a test session needs to be created for each of the supported browsers. Results should be added to each of the conditions in the created test sessions with a pass `(/)` or fail `(x)` mark.
+
+Any issues found during testing should be raised as a "Defect" in Jira unless the ticket is already of a “Bug” type, in which case comments and screenshots/recordings/GIFs can be added to the existing ticket.
```

Comment:
I think we should mention that if the bug is present in master and not a defect of this ticket, it should be its own `bug` ticket and shouldn't block this ticket.

Reply:
At line 80 I state "If the issue is present on the master branch you should first check if the issue has already been logged in GitHub. If it is present, it should be upvoted with a “+1” comment...Collaborators internal to Sage should also log a Jira ticket for the issue as a "Bug" type with the GitHub issue linked." I'll amend the sentence to read "Collaborators internal to Sage should also log a **_separate_** Jira ticket for the issue, **_not blocking this one_**, as a "Bug" type...". to specify that.